Add DMA driver support for PIC32CX SG boards

ArunMCHP opened the pull request:
This PR adds the DMA controller node for the PIC32CX SG family and updates the corresponding PIC32CX-SG61 board device tree to enable DMA support.

ArunMCHP committed:
dts: arm: microchip: add DMA node for PIC32CX-SG family devices

Add DMA controller node to enable DMA G1 driver
support on PIC32CX-SG devices.

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
soc: microchip: pic32cx_sg: add DT helper file

Add DT helper file for PIC32CX-SG family devices to support
SoC-specific device tree definitions and utilities.

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
boards: microchip: pic32cx_sg61_cult: add DMA to supported list

Update pic32cx_sg61_cult.yml to include DMA in the supported
features list.
Also enable the DMA controller node in pic32cx_sg61_cult.dts.

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

nandojve commented on soc/microchip/pic32c/pic32cx_sg/common/mchp_dt_helper.h:
```diff
@@ -0,0 +1,30 @@
+#define MCHP_DT_INST_DMA_CHANNEL(n, name) MCHP_DT_INST_DMA_CELL(n, name, channel)
+#define MCHP_DT_INST_DMA_CTLR(n, name)                                                             \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, dmas),					\
+			(DT_INST_DMAS_CTLR_BY_NAME(n, name)),				\
```
use tabs

```c
#define MCHP_DT_INST_DMA_CELL(n, name, cell)						\
	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, dmas),					\
			(DT_INST_DMAS_CELL_BY_NAME(n, name, cell)),			\
			(0xff))
#define MCHP_DT_INST_DMA_TRIGSRC(n, name) MCHP_DT_INST_DMA_CELL(n, name, trigsrc)
#define MCHP_DT_INST_DMA_CHANNEL(n, name) MCHP_DT_INST_DMA_CELL(n, name, channel)
#define MCHP_DT_INST_DMA_CTLR(n, name)							\
	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, dmas),					\
			(DT_INST_DMAS_CTLR_BY_NAME(n, name)),				\
```

nandojve reviewed (changes requested):
you need enable all the dma tests in this PR.

nandojve commented:
Hi @ArunMCHP ,

> > you need enable all the dma tests in this PR.
> 
> We plan to add the test overlay files in separate PR.

When we bring the functionality we should bring the tests. DMA usually causes a _lot of headaches_ because most of is in tree do not uses by default. So, it is even more important in this case the tests.

ArunMCHP committed:
tests: dma: microchip: Add board overlay file

Add board overlay file for pic32cx_sg61_cult to enable dma
test cases to run on this board

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
tests: dma: microchip: Add board overlay file

Add board overlay file for pic32cx_sg41_cult to enable dma
test cases to run on this board

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
boards: microchip: pic32cx_sg41_cult: add DMA to supported list

Update pic32cx_sg41_cult.yml to include DMA in the supported
features list.
Also enable the DMA controller node in pic32cx_sg41_cult.dts.

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>
